# Principal Component Analysis (PCA)

In [5]:
%matplotlib inline

from IPython.display import display
import pandas as pd

For this notebook we will be using bike sharing data from the city of Toulouse.

In [7]:
df = pd.read_csv('data/bike_stations.csv', index_col=0)
display(df)

0-0       1-0       2-0       3-0  \
00189 - MURET BOUTINON           0.890442  0.783402  0.754204  0.694712   
00157 - U.R.S.S  PEYROUSET       0.794423  0.694662  0.666052  0.611976   
00069 - FEUGA                    0.237852  0.246162  0.339617  0.288618   
00181 - PLANA GARNIER            0.677826  0.659270  0.711563  0.626856   
00014 - PERIGORD                 0.377450  0.221058  0.165181  0.175972   
00200 - PLACE DE L'ORMEAU        0.280784  0.262712  0.249726  0.254810   
00125 - BARRIÈRE DE PARIS        0.628550  0.538059  0.426214  0.447657   
00110 - CABANEL - SERRES         0.355757  0.460227  0.449365  0.484444   
00145 - GOUNOD - GARROS          0.782466  0.828224  0.896189  0.828138   
00132 - MINIMES BILARD           0.657230  0.581545  0.530487  0.489093   
00278 - ARENES ROMAINE           0.348783  0.248060  0.179980  0.160522   
00211 - LE DORMEUR/ LA TERRASSE  0.274452  0.329444  0.354964  0.305431   
00232 - NARBONNE/ MIROUSE        0.720004  0.758865  0.518701  0.699634   
00275 - PLACE DES PRADETTES      0.324028  0.327882  0.278337  0.301349   
00119 - CHAPOU                   0.567437  0.836700  0.807039  0.764906   
00034 - SAINT LAZARE             0.172406  0.167258  0.134010  0.134013   
00116 - ST MICHEL NOTRE DAME     0.555604  0.784162  0.694493  0.636504   
00265 - PASSAGE BORDELONGUE      0.407429  0.390135  0.374979  0.386021   
00113 - MUSEUM LAMARCK           0.466372  0.501623  0.485037  0.462160   
00267 - GONIN                    0.539320  0.562415  0.443002  0.463300   
00246 - WAGNER / JEAN BRUNHES    0.855428  0.956322  0.908525  0.829668   
00107 - SEBASTOPOL               0.580208  0.405424  0.363275  0.425636   
00059 - CONCORDE                 0.488489  0.844968  0.828578  0.827910   
00149 - RAISIN MINIMES           0.430146  0.486026  0.626398  0.572974   
00078 - PLACE INT ST CYP. DARRE  0.594896  0.467537  0.466940  0.510036   
00209 - BAROUX/ DARASSE          0.071195  0.161265  0.190668  0.130473   
00220 - TABAR /UNIVERSITE        0.336865  0.294080  0.272398  0.258707   
00027 - QUAI DE TOUNIS           0.269148  0.229562  0.370590  0.348655   
00256 - PLACE DE LA RONTONDE     0.528541  0.469276  0.452564  0.427624   
00264 - LAFOURGUETTE             0.644819  0.502137  0.466620  0.499469   
...                                   ...       ...       ...       ...   
00135 - BONNEFOY   LEMAITRE      0.579271  0.645983  0.657114  0.711002   
00166 - RANGUEIL GUYOU           0.414743  0.262517  0.274561  0.318290   
00175 - METRO MAUNOURY           0.376489  0.342397  0.307224  0.310130   
00048 - PLACE DU SALIN           0.218936  0.129342  0.170876  0.212331   
00192 - RTE DE ST SIMON/PRADO    0.645330  0.745018  0.672889  0.679305   
00234 - LOIN / RAYNAL            0.788781  0.799458  0.800285  0.818472   
00259 - PLACE RICHE              0.699345  0.688106  0.603256  0.631763   
00007 - PLACE VICTOR HUGO        0.473786  0.425590  0.449653  0.427925   
00001 - POIDS DE L'HUILE         0.256397  0.080524  0.069139  0.068083   
00239 - LESPINET / HERBETTES     0.776992  0.755576  0.728458  0.766254   
00121 - SERRES - A. BERNARD      0.211593  0.381377  0.455219  0.381909   
00087 - LASCROSSES LECLERC       0.288306  0.196342  0.152886  0.333006   
00053 - PLACE SAINT JULIEN       0.166469  0.142113  0.187318  0.188297   
00248 - BOUQUETINS/E. ROSTAND    0.701325  0.662580  0.668977  0.695674   
00081 - JARDIN RAYMOND VI        0.764456  0.764243  0.765822  0.757279   
00139 - BARRIERE DE LOMBEZ       0.719363  0.619798  0.491854  0.453244   
00093 - PRINTEMPS                0.464028  0.784854  0.804598  0.805548   
00003 - POMME                    0.301799  0.096785  0.114830  0.092982   
00040 - COLOMBETTE               0.526965  0.583085  0.463639  0.411615   
00144 - SUISSE - LAUSANE         0.763868  0.776851  0.762324  0.725463   
00137 - BRUNHES-FONTAINES        0.747465  0.904244  0.824829  0.848606   
00174 - MAUNOURY / MELA          0.619480  0.687977  0.634823  

Each cell contains the average $\frac{bikes}{bikes + spaces}$ ratio. Each column name is a weekday-hour pair (with weekdays starting at 0). For example, the *00189 - Muret Boutinon* station is on average 89% full on Mondays at midnight (first row, first column). 

The last three columns contain information about the location of each station (latitude, longitude and altitude).

ImportError: No module named 'prince'